File: 0020. Valid Parentheses/00020-Valid-Parantheses.cpp
class Solution {
public:
    bool isValid(string s) {
        if (s.length() % 2 != 0) {
            return false;
        }

        int size = 0;
        char stack[s.length()];
        for (int i = 0; i < s.length(); i++) {
            char ch = s[i];
            if (ch == '(') {
                stack[size] = ')';
                size++;
            } else if (ch == '{') {
                stack[size] = '}';
                size++;
            } else if (ch == '[') {
                stack[size] = ']';
                size++;
            } else if (size == 0 || ch != stack[size - 1]) {
                return false;
            } else {
                size--;
            }
        }
        return size == 0;
    }
};

File: 0020. Valid Parentheses/00020-Valid-Parantheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        for bracket in s:
            if bracket in '[({':
                stack.append(bracket)
            else:
                if not stack:
                    return False
                popped = stack.pop()
                if popped == '[' and bracket != ']':
                    return False
                elif popped == '(' and bracket != ')':
                    return False
                elif popped == '{' and bracket != '}':
                    return False
                else:
                    continue
        
        return True if not stack else False